internal/jimm/juju/migrationtarget.go

```diff
@@ -413,25 +419,35 @@ func (j *JujuManager) Import(ctx context.Context, user *openfga.User, serialized
 	if err != nil {
 		return errors.E(op, fmt.Errorf("failed to deserialize model description: %w", err))
 	}
+
 	incomingMigration := &dbmodel.IncomingModelMigration{
 		ModelUUID: sql.NullString{
 			String: modelDescription.Tag().Id(),
 			Valid:  true,
 		},
 	}
+
 	err = j.Database.GetIncomingModelMigration(ctx, incomingMigration)
 	if err != nil {
 		return errors.E(op, fmt.Errorf("failed to add incoming model migration: %w", err))
 	}
+
 	err = j.modifyModelDescription(modelDescription, incomingMigration.UserMapping)
 	if err != nil {
 		return errors.E(op, fmt.Errorf("failed to modify model description: %w", err))
 	}
-	err = j.importModelFromDescription(ctx, incomingMigration.TargetController.ID, modelDescription)
+
+	model, offers, err := j.importFromDescription(ctx, incomingMigration.TargetController.ID, modelDescription)
 	if err != nil {
 		return errors.E(op, fmt.Errorf("failed to import model from description: %w", err))
 	}
 
+	err = j.addModelAndOfferPermissions(ctx, user, model, offers)
+	if err != nil {
+		return errors.E(op, fmt.Errorf("failed to add resource permissions: %w", err))
```

Comment:
now, this is a bit of a tricky situation.. we've imported the model into JIMM's state, but failed to set permissions. how do we get out of this situation? should we give users the tooling to re-set model and offer permissions? but the problem is that we don't have a record of who should have permission to them..

Reply:
I think this might be okay, if this fails, the migration will be aborted and in JIMM's Abort handler we remove the model.